platform/ui/src/contextProviders/ViewportGridProvider.jsx

```diff
@@ -69,12 +69,13 @@ export function ViewportGridProvider({ children, service }) {
     [dispatch]
   );
   const setDisplaysetForViewport = useCallback(
-    ({ viewportIndex, displaySetInstanceUID }) =>
+    ({ viewportIndex, displaySetInstanceUID, frameIndex }) =>
       dispatch({
         type: 'SET_DISPLAYSET_FOR_VIEWPORT',
         payload: {
           viewportIndex,
           displaySetInstanceUID,
+          frameIndex
```

Comment:
frameIndex, but no `SOPInstanceUID`/`ImageId`/`imageIndex`?

Reply:
ImageIndex =)

Reply:
@JamesAPetts, want to clarify that this makes sense.

If you have a displaySet, and you have its list of images, this is comparable to a csTools "stack state". Each image in the displaySet _could_ be a frame, but we don't care. What we're seeking for, and trying to display, is a specific image that is indexed in the displaySet.

Saying all of this out loud, so anyone can push on my understanding. If the above is true, then `imageIndex` as a solution for and viewport that is a 2D stack should work, I believe. Maybe different for volumes? But we can cross that bridge later.